SingularityCE: Adding glib as a dependency

usivakum opened the pull request:
As per the [SingularityCE Installation Guide](https://docs.sylabs.io/guides/main/admin-guide/installation.html#id1), `glib2` is a dependency, which was previously missing

usivakum committed:
adding glib as a dependency for singularityce

haampie commented:
Somehow it seems conmon is bundled as a binary, so adding it as a dependency is not helpful; it picks up system glib.

Can you also add `squashfs` as a run-type dep?

```
checking: mksquashfs... yes (/usr/bin/mksquashfs)
```

usivakum committed:
added conmon and squashfs as dependency

usivakum commented:
@haampie Thanks for the pointers! I have added `conmon` as a dependency which resolves the `glib` issue. I have also added `squashfs` as a dependency as suggested

wdconinc commented:
This has caused a regression in apptainer, which does not like the `--without-conmon` flag in its [mconfig](https://github.com/apptainer/apptainer/blob/main/mconfig), since apptainer uses [singularityce]([url](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/apptainer/package.py)) as its base.
